### PR TITLE
[PCL] set QHULL_FOUND

### PR DESCRIPTION
**Pull request by JavierMatosD**
This patch is a workaround to resolve https://github.com/microsoft/vcpkg/discussions/27573.

PCL build system looks for QHULL_FOUND: https://github.com/PointCloudLibrary/pcl/blob/f36a69a5e89953708990c4669317f989d532cf08/PCLConfig.cmake.in#L312

However, find_package() sets Qhull_FOUND. 
https://github.com/PointCloudLibrary/pcl/blob/f36a69a5e89953708990c4669317f989d532cf08/PCLConfig.cmake.in#L139

Hence, PCL doesn't find QHULL_FOUND and disables it.

The reason other PCL customers are not broken is that they set it explicitly https://github.com/PointCloudLibrary/pcl/blob/f36a69a5e89953708990c4669317f989d532cf08/cmake/Modules/FindQhull.cmake#L77 
and we delete their find module: https://github.com/JavierMatosD/vcpkg/blob/f4b909297ac33cc2c966c4d914b65b99903dabdb/ports/pcl/portfile.cmake#L25

Note: Updating the port will likely fix this issue. However, this patch should unblock @drathin4 in the meantime.

### CODE REVIEW

**Comment by dg0yt** [comment]
Hm, there is already fix-find-qhull.patch for the import, and this adds a separate patch for the export. It is getting hard to maintain the pile of patches.
I don't see enough context of the new export patch, but the import patch could be much simplified if using `find_package(QHULL names Qhull REQUIRED), to directly get Qhull. The exported config would need a similiar `find_dependency` but I can't see it in the patches.

**Comment by Osyotr** [comment]
Qhull search is probably not the only thing that needs fixing:
```
1> [CMake] ** WARNING ** io features related to png will be disabled
1> [CMake] ** WARNING ** surface features related to qhull will be disabled
```

**Comment by dg0yt** [comment]
... that's why it is important to ensure early that the original `<Pkg>_FOUND` is set (one local change), instead of patching it out where it is used (N changes in M files).

**Comment by JavierMatosD** [comment]
> Hm, there is already fix-find-qhull.patch for the import, and this adds a separate patch for the export. It is getting hard to maintain the pile of patches. I don't see enough context of the new export patch, but the import patch could be much simplified if using `find_package(QHULL names Qhull REQUIRED)`, to directly get `QHULL_FOUND`. The exported config would need a similiar `find_dependency` but I can't see it in the patches.

Upstream had a release last year that should fix some of these issues and reduce the number of patch files altogether. This patch is just to unblock another user while I comb through all the patches to see what is still needed when I update the port.

**Comment by JavierMatosD** [comment]
> Qhull search is probably not the only thing that needs fixing:
> 
> ```
> 1> [CMake] ** WARNING ** io features related to png will be disabled
> 1> [CMake] ** WARNING ** surface features related to qhull will be disabled
> ```

You are correct. I am also looking into why png is being disabled, but I will probably just update the port first and see if the problem persists.

**Comment by BillyONeal** [comment]
> I don't see enough context of the new export patch, but the import patch could be much simplified if using `find_package(QHULL names Qhull REQUIRED)`, to directly get `QHULL_FOUND`. The exported config would need a similiar `find_dependency` but I can't see it in the patches.

To clarify, you mean like this? https://github.com/BillyONeal/vcpkg/commit/f966f49747850a2f2b63d3f22978fbcb33080242

**Comment by dg0yt** [comment]
> To clarify, you mean like this? [BillyONeal@f966f49](https://github.com/BillyONeal/vcpkg/commit/f966f49747850a2f2b63d3f22978fbcb33080242)

Yes  Note that `NAMES` implies Config mode. This is already used in other ports.

**Review by BillyONeal — APPROVED** [review]
I'm happy if @dg0yt is happy :)

File: ports/pcl/fix-find-qhull.patch [hunk 1]
```diff
@@ -10,30 +10,40 @@ index a1d9bb5..73b1465 100644
...
-   endif()
+@@ -187,6 +187,11 @@ if(VTK_FOUND)
  endif()
  
--if(QHULL_FOUND)
-+if(Qhull_FOUND)
+ if(QHULL_FOUND)
 +  if(QHULL_USE_STATIC)
 +    set(QHULL_LIBRARIES Qhull::qhullcpp Qhull::qhullstatic Qhull::qhullstatic_r)
 +  else()
```
Comment (dg0yt) [inline ports/pcl/fix-find-qhull.patch:49]: Another post-merge comment from the land of happiness:
I would prefer to see this moved next to `find_package(QHull ...)`. Resulting in a single spot to patch  for dependency import in order to simulate upstream's Find module interface.

**Pull request merged**